app/services/hyrax/work_uploads_handler.rb

```diff
@@ -114,10 +115,12 @@ def make_file_set_and_ingest(file)
     # @api private
     #
     # @todo figure out how to know less about Work's ideas about FileSet use here. Maybe post-Wings, work.
-    def append_to_work(file_set)
+    def append_to_work(file_set, user)
       work.member_ids << file_set.id
       work.representative_id = file_set.id if work.respond_to?(:representative_id) && work.representative_id.blank?
       work.thumbnail_id = file_set.id if work.respond_to?(:thumbnail_id) && work.thumbnail_id.blank?
+      @persister.save(resource: work)
+      Hyrax.publisher.publish('object.metadata.updated', object: work, user: user)
```

Comment:
should it be a `file.set.attached` event instead?

Reply:
`#on_file_set_attached` initiates `FileSetAttachedEventJob` for a FileSet.  That job creates an event notification that the `FileSet` was attached to the work.  It doesn't reindex the work.

So, I think `file.set.attached` should be published for each file.  And the `object.metadata.updated` should be done once for the work.

Reply:
The file.set.attached event is already being initiated.  The only change needed was to move `object.metadata.updated` for the work outside the files loop block so the indexing happens only once.